app/controllers/documents_controller.rb

```diff
@@ -125,4 +134,37 @@ def prepare_for_mobile
     session[:mobile_param] = params[:mobile] if params[:mobile]
    # request.format = :mobile if mobile_device?
   end
+
+  private
+
+  def catalog_texts
+
+    if catalogue_enabled?
+       status, results = Melcatalog.texts
+       return results[:text] unless results[:text].nil?
+    end
+    return []
+  end
+
+  def catalog_content( doc )
+
+    if catalogue_enabled?
+      # we put placeholder content in earlier and replace with the real thing now
+      if doc.text.start_with?( "EID:" )
+         eid = doc.text.split( ":",2 )[ 1 ]
+         status, entry = Melcatalog.get( eid, 'stripxml' )
+         if status == 200 && entry && entry[:text] && entry[:text][ 0 ] && entry[:text][ 0 ]['content']
+           doc.text = entry[:text][ 0 ]['content']
+         else
+           doc.text = "Error getting document content from the catalog; status = #{status}, eid = #{eid}"
+         end
+      end
+    end
+  end
+
+  # helper to determine if we should support content from the MEL catalog
+  def catalogue_enabled?
```

Comment:
True, that could work. I like breaking out the full configuration, since that affords slightly more flexibility to the implementor/installer. But I agree with you, this approach to config is getting unwieldy, and calls for a rewrite. This approach lets us accept the pull request for now, and move ahead, and I will put a note into our tracker that we need to attend to configuration globally soon. Do you think I could ask you to make those changes, and push the commits to your PR, since I can't?